src/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,332 @@
+  placeholder,
+  customClassName,
+  position = "bottom",
+  role,
+  isMultiSelect = false,
+  options,
+  selectedOption,
+  onSelect,
+  canSelectAlreadySelected = false,
+  hasHeaderBoxShadow = false,
+  hasDeselectOption = true,
+  hasError,
+  isDisabled,
+  isMenuOpenHook,
+  headerWithoutButton,
+  shouldCloseOnSelect = true,
+  onMenuVisibilityChange,
+  areOptionsFetching = false,
+  deselectOptionTitle = "",
+  noOptionsMessage,
+  headerButtonClassName = "white-button",
+  shouldJumpToQuery = true,
+  shouldShowEmptyOptions = true,
+  canOpenDropdownMenu = true
+}: DropdownProps<OptionIdShape>) {
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const [computedOptions, setComputedOptions] = useState<DropdownOption[]>([]);
+  const defaultMenuOpenHook = useState(false);
+  const [isMenuOpen, setMenuVisibility] = isMenuOpenHook || defaultMenuOpenHook;
+  const [isMouseDown, setIsMouseDown] = useState(false);
+  const [keyQuery, setKeyQuery] = useState("");
+  const [focusedOptionIndex, setFocusedOptionIndex] = useState(
+    generateInitialFocusedDropdownOptionIndex(position, computedOptions, selectedOption)
+  );
+  // Button will gain focus when shouldFocusOnButton is set to true
+  const [shouldFocusOnHeaderButton, setShouldFocusOnHeaderButton] = useState(false);
+  const containerClassName = classNames("dropdown-container", customClassName, {
+    "dropdown-open": isMenuOpen,
+    "option-selected": Boolean(selectedOption || deselectOptionTitle),
+    "has-header-box-shadow": hasHeaderBoxShadow,
+    "has-error": hasError,
+    disabled: isDisabled || areOptionsFetching
+  });
+  const shouldShowMenu = isMenuOpen && (shouldShowEmptyOptions || options.length > 0);
+
+  useEffect(() => {
+    setComputedOptions(
+      options.length && hasDeselectOption
+        ? [{...DESELECT_OPTION, title: deselectOptionTitle}, ...options]
+        : options
+    );
+  }, [options, deselectOptionTitle, hasDeselectOption]);
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
+
+    if (keyQuery && isMenuOpen) {
+      const closestMatchingItemIndex = findIndexForClosestMatch(
+        computedOptions,
+        keyQuery
+      );
+
+      if (closestMatchingItemIndex > -1) {
+        setFocusedOptionIndex(closestMatchingItemIndex);
+      }
+
+      timeoutId = setTimeout(() => {
+        setKeyQuery("");
+      }, DROPDOWN_SEARCH_QUERY_TIMEOUT_IN_MS);
+    }
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [keyQuery, computedOptions, isMenuOpen]);
+
+  const defaultDropdownHeader = (
+    <div className={"dropdown-header-button-default-content"}>
+      {selectedOption && selectedOption.icon}
+
+      <span className={"dropdown-header-button-text"}>
+        {selectedOption ? selectedOption.title : deselectOptionTitle || placeholder}
+      </span>
+
+      {areOptionsFetching ? (
+        <Spinner spinnerColor={"black"} backgroundColor={"#EBEBEB"} />
+      ) : (
+        <CaretDownIcon aria-hidden={true} />
```

Comment:
Let's add a className to the svg icon

Reply:
`react-svg-loader` doesn't accept `className` currently